File: Data Structures & Algorithms/two-integer-sum-ii/submission-7.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        # array is sorted increasing order
        # return 1 indexed soltiion of i1 and i2 so they ad up to the target
        # i1 < i2

        n = len(numbers)
        l = 0
        r = n - 1

        while l < r:
            
            # have the value
            value = numbers[l] + numbers[r]

            if value > target:
                r -= 1
            elif value < target: 
                l += 1
            else: # we found the target
                return [l + 1, r + 1]

            

        

File: Data Structures & Algorithms/binary-search/submission-7.cs
public class Solution {
    public int Search(int[] nums, int target) {

        int l = 0;
        int r = nums.Length - 1;

        while (l <= r)
        {
            int mid = (l + r) / 2;
            Console.WriteLine($"l = {l} | m = {mid} | r = {r}");

            if (nums[mid] < target)
                l = mid + 1;
            else if (nums[mid] > target)
                r = mid - 1;
            else
                return mid;
        }
        

        // Not found so return -1
        return -1;
    }
}
